From 361c530ef4cb1e3b63162cd873c7202f55a71300 Mon Sep 17 00:00:00 2001
From: ndifolcx <neilx.di.folco@intel.com>
Date: Fri, 30 Nov 2012 14:21:44 +0100
Subject: Support new audio devices enum values introduced in JB MR1 code drop

BZ: 71815
[CLEAN]

Today, each audio device is identified by a bit in a 32 bit field with
16 bits for output devices and 16 bits for input devices.
So the number of output devices is now exhausted i.e. we can't add new output devices.

So in order to add new devices, a new convention for audio device
identification is defined, allowing 31 different input and output devices.
Bit 31 of the bit field is reserved and when set indicates an input device.

Category: aosp improvement
Domain: AudioComms-AudioPolicy
Origin: internal
Upstream-Candidate: yes
Change-Id: Iceae9d709354b69ba51d0465b53681058b42b0b8
Orig-Change-Id: I9b32706076fe86268cb1501ba35f517702fc3e15
Signed-off-by: ndifolcx <neilx.di.folco@intel.com>
---
 include/hardware_legacy/AudioSystemLegacy.h |   10 ++--------
 1 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/include/hardware_legacy/AudioSystemLegacy.h b/include/hardware_legacy/AudioSystemLegacy.h
index eaaa982..0b200da 100644
--- a/include/hardware_legacy/AudioSystemLegacy.h
+++ b/include/hardware_legacy/AudioSystemLegacy.h
@@ -324,16 +324,10 @@ public:
 
 #if 1
     static bool isOutputDevice(audio_devices device) {
-        if ((popcount(device) == 1) && ((device & ~DEVICE_OUT_ALL) == 0))
-             return true;
-         else
-             return false;
+        return audio_is_output_device((audio_devices_t)device);
     }
     static bool isInputDevice(audio_devices device) {
-        if ((popcount(device) == 1) && ((device & ~DEVICE_IN_ALL) == 0))
-             return true;
-         else
-             return false;
+        return audio_is_input_device((audio_devices_t)device);
     }
     static bool isA2dpDevice(audio_devices device) {
         return audio_is_a2dp_device((audio_devices_t)device);
-- 
1.7.4.1

